server/s3_common_utilities.cc

```diff
@@ -179,3 +182,29 @@ std::string evhtp_error_flags_description(uint8_t errtype) {
 }
 
 }  // namespace S3CommonUtilities
+
+void s3_kickoff_graceful_shutdown(int ignore) {
+  extern int global_shutdown_in_progress;
+  extern evbase_t *global_evbase_handle;
+
+  if (!global_shutdown_in_progress) {
+    global_shutdown_in_progress = 1;
+    // signal handler
+    S3Option *option_instance = S3Option::get_instance();
+    int grace_period_sec = option_instance->get_s3_grace_period_sec();
+    struct timeval loopexit_timeout = {.tv_sec = 0, .tv_usec = 0};
+
+    // trigger rollbacks & stop handling new requests
+    option_instance->set_is_s3_shutting_down(true);
+
+    if (grace_period_sec > 6) {
```

Comment:
what is this "6" can we make it constant variable ?

Reply:
It was deleted